src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1738,5 +1776,217 @@ .maxstack  1
             verifier.VerifyMissing("ThrowIfNull");
             verifier.VerifyMissing("Throw");
         }
+
+        [Fact]
+        public void UserDefinedConversion()
+        {
+            var source = @"
+class C
+{
+    public static bool operator ==(C c1, C c2) => false;
+    public static bool operator !=(C c1, C c2) => false;
+    public override bool Equals(object obj) => false;
+    public override int GetHashCode() => 0;
+
+    public static void M(C c!!)
+    {
+        if (c == null) { System.Console.Write(1); }
+    }
+}
+";
+            var verifier = CompileAndVerify(source);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.M", @"
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldstr      ""c""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ldarg.0
+  IL_000c:  ldnull
+  IL_000d:  call       ""bool C.op_Equality(C, C)""
+  IL_0012:  brfalse.s  IL_001a
+  IL_0014:  ldc.i4.1
+  IL_0015:  call       ""void System.Console.Write(int)""
+  IL_001a:  ret
+}");
+        }
+
+        [Fact]
+        public void Override_NullCheckedDifference()
+        {
+            var source =
+@"
+class Base
+{
+    public virtual void M1(string x!!) { }
+    public virtual void M2(string x) { }
+}
+
+class Derived : Base
+{
+    public override void M1(string x) { }
+    public override void M2(string x!!) { }
+}";
+            var verifier = CompileAndVerify(source);
+            verifier.VerifyDiagnostics();
```

Comment:
Not sure if it does make sense to also run this test with NRT enabled and still ensure no nullable-related warnings are produced?

Reply:
Sounds good. Done.